browser/src/control/jsdialog/Widget.TreeView.ts

```diff
@@ -1326,6 +1326,16 @@ class TreeViewControl {
 			}
 
 			preventDef = true;
+		} else if (event.key === 'Tab') {
+			// Ensure focus change is completed
+			setTimeout(() => {
```

Comment:
We should try to avoid using timeouts when possible to avoid potential races.
Did you consider using `lost focus` event?